pkg/autoscaler/autoscaler.go

```diff
@@ -437,12 +439,12 @@ var _ = Describe("Autoscaler should", framework.LabelAutoscaler, Serial, func()
 				transientMachineSets[i] = machineSet
 			}
 
-			// balance similar nodes requires that all the participating MachineSets have the same
+			// balance similar nodes requires that all the participating Nodes have the same
 			// instance types and labels. while the instance types should be the same, we want to
 			// ensure that no extra labels have been added.
 			// TODO it would be nice to check instance types as well, this will require adding some deserialization code for the machine specs.
-			By("Ensuring both MachineSets have the same labels")
-			Expect(reflect.DeepEqual(transientMachineSets[0].Labels, transientMachineSets[1].Labels)).Should(BeTrue())
+			By("Ensuring both MachineSets have the same .spec.template.spec.labels")
+			Expect(reflect.DeepEqual(transientMachineSets[0].Spec.Template.Spec.Labels, transientMachineSets[1].Spec.Template.Spec.Labels)).Should(BeTrue())
```

Comment:
Should this expectation have a context added, or be reworked to compare the two maps?
```suggestion
			Expect(transientMachineSets[0].Spec.Template.Spec.Labels).To(Equal(transientMachineSets[1].Spec.Template.Spec.Labels), "Test machine sets should have the same labels")
```

Reply:
sounds good to me